### PR TITLE
deprecate ignore_version_mismatch

### PR DESCRIPTION
**Pull request by braingram**
This PR deprecates `ignore_version_mismatch`. Since asdf 3.0.0 it has done nothing (it was only used for the legacy extension API which was removed in asdf 3.0.0).

There are a few downstream uses in stdatamodels mostly in tests. However there is one jwst test that fails due to a non-specific warning filter:
https://github.com/spacetelescope/jwst/blob/89f126a035751674f29a9027e2deae11b3e63fca/jwst/assign_wcs/tests/test_schemas.py#L57

Requires https://github.com/spacetelescope/stdatamodels/pull/313
and likely https://github.com/spacetelescope/jwst/pull/8693 (or a release of stdatamodels with the above PR)
before merging. I'm opening it for review now in case there is discussion about the warning message (changes to the message would require updating the jwst PR).

Closes https://github.com/asdf-format/asdf/issues/1812

# Checklist:

- [x] pre-commit checks ran successfully
- [x] tests ran successfully
- [x] for a public change, a changelog entry was added
- [ ] for a public change, documentation was updated
- [ ] for any new features, unit tests were added

### CODE REVIEW

File: asdf/_asdf.py
Comment (zacharyburnett) [inline asdf/_asdf.py:171]:
```suggestion
                "ignore_version_mismatch is deprecated and has no effect on `asdf>=3.0.0`",
```

File: asdf/_asdf.py
Comment (zacharyburnett) [inline asdf/_asdf.py:105]:
```suggestion
            Deprecated and unused. This setting has no effect on `asdf>=3.0.0`
```

File: asdf/_asdf.py
Comment (zacharyburnett) [inline asdf/_asdf.py:1662]:
```suggestion
        Deprecated and unused. This setting has no effect on `asdf>=3.0.0`
```

File: pytest_asdf/plugin.py
Comment (zacharyburnett) [inline pytest_asdf/plugin.py:290]:
```suggestion
            "asdf_schema_ignore_version_mismatch is deprecated and has no effect on `asdf>=3.0.0`",
```

File: asdf/_tests/test_deprecated.py
Comment (zacharyburnett) [inline asdf/_tests/test_deprecated.py:153]:
```suggestion
    with pytest.warns(AsdfDeprecationWarning, match="ignore_version_mismatch is deprecated"):
```

**Review by zacharyburnett — APPROVED** [review]
LGTM, one suggestion for a slightly more explicit warning message

**Comment by braingram** [comment]
Thanks for taking a look.

Unfortunately the jwst PR was merged :-/ That's my bad for not waiting to open that one. It includes a warning ignore that matches on the message currently used in this PR.
https://github.com/spacetelescope/jwst/pull/8693/files#diff-3f8cadc9ec467d860d9ac11aa3e65b4f126e2c44da2e9f45692ea9917ac58526R64

We could:
- make another jwst PR with the updated warning message
- merge this PR (once the new jwst PR is merged and the stdatamodels PR is approved and merged)

Or if you think the message is "good enough" we could go ahead with the existing message. We'd still have to wait for the stdatamodels PR so making another jwst PR doesn't seem like an issue. Would you like me to update the warning filter in jwst?

**Comment by zacharyburnett** [comment]
> Thanks for taking a look.
> 
> Unfortunately the jwst PR was merged :-/ That's my bad for not waiting to open that one. It includes a warning ignore that matches on the message currently used in this PR. https://github.com/spacetelescope/jwst/pull/8693/files#diff-3f8cadc9ec467d860d9ac11aa3e65b4f126e2c44da2e9f45692ea9917ac58526R64
> 
> We could:
> 
> * make another jwst PR with the updated warning message
> * merge this PR (once the new jwst PR is merged and the stdatamodels PR is approved and merged)
> 
> Or if you think the message is "good enough" we could go ahead with the existing message. We'd still have to wait for the stdatamodels PR so making another jwst PR doesn't seem like an issue. Would you like me to update the warning filter in jwst?

nah, I think this message is good enough!

**Pull request merged**
